### PR TITLE
feat: PrivateLink support

### PR DESCRIPTION
**Pull request by wandb-zacharyblasczyk**
This pull request introduces a p`rivate_link module` to enable to automatic setting up AWS PrivateLink. 

The key variable is `private_link_allowed_account_ids`, which triggers the module only when a list of allowed AWS account IDs is provided. 

Other crucial variables include `namespace` for resource naming (and the limit of LB's names), `network_private_subnets` for specifying VPC private subnets and `deletion_protection` for enabling deletion protection.

### CODE REVIEW

File: modules/private_link/main.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+resource "aws_lb_target_group_attachment" "nlb_to_alb" {
+  target_group_arn = aws_lb_target_group.nlb_tg.arn
+  target_id        = data.aws_lb.alb.arn
+}
+
+resource "aws_vpc_endpoint_service" "private_link" {
+  acceptance_required        = false
+  network_load_balancer_arns = [aws_lb.nlb.arn]
+  allowed_principals         = [for id in var.allowed_account_ids : "arn:aws:iam::${id}:root"]
```
Comment (jsbroks) [inline modules/private_link/main.tf:45]: should we be using root here?

File: modules/private_link/main.tf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+resource "aws_lb" "nlb" {
+  name                       = local.lb_name_truncated
+  internal                   = true
+  load_balancer_type         = "network"
+  subnets                    = var.network_private_subnets
+  enable_deletion_protection = var.deletion_protection
+}
+
+resource "aws_lb_target_group" "nlb_tg" {
```
Comment (jsbroks) [inline modules/private_link/main.tf:14]:
```suggestion
resource "aws_lb_target_group" "nlb" {
```

**Review by jsbroks — COMMENTED** [review]
amazing. Just get back to me on the root and Ill approve

File: modules/private_link/main.tf (hunk 1 above)
Comment (wandb-zacharyblasczyk) [reply modules/private_link/main.tf:45]: Looking at what we have in prod for another account, yes.

**Review by jsbroks — APPROVED** [review]

**Pull request merged**